### ISSUE TITLE
Can I use an English dataset for this repo?

### ISSUE THREAD

**Issue by Shaobo-Z** [khanld/ASR-Wav2vec-Finetune#7]
In the source code, you used Vietnamese for training and validation. If I want to fine-tune a model that is in English and has English dataset, is there anything that I should change?

**Comment by khanld** [comment]
No, you just have to prepare the English dataset

**Comment by Shaobo-Z** [comment]
This is how my dataset looks like ⬇
![image](https://github.com/khanld/ASR-Wav2vec-Finetune/assets/38678756/aacb5306-c4e4-4144-9540-4749ff7b9e74)

And this is what I got ⬇. There is changes with train_loss, train_lr,..... However, the train_wer is always 1.0000. 
![image](https://github.com/khanld/ASR-Wav2vec-Finetune/assets/38678756/78b823ad-f2fb-4d00-b0d6-1ac5d1ba6df7)

Checked:
1. Sample Rate: by using `librosa.get_samplerate`. I got 16000. 
2. Transcript is correct.
3. Only modify the file_path and iteration in the configure file.
4. The pre-trained model is facebook/wave2vec2-base. 

I tried multiple ways. However, the result remains the same. Any ideas? Plz.

**Comment by khanld** [comment]
I can see that your dataset is relatively small, so the number of update steps per epoch is only 5. Have your try a longer run and check if the behavior remains. Take a look at the vocab.json file whether it contains the correct English characters.

**Comment by ghosthunterk** [comment]
Encountered the same problem even with larger dataset (91 steps and 20 epochs).

**Comment by khanld** [comment]
I have not tried on other language datasets yet. Can you share more information about your dataset, config, tensorboard,…

**Comment by ghosthunterk** [comment]
Python 3.8
Pip install all in requirements.txt, with exception of torch 1.7.1 i had to use (conda install pytorch==1.7.1 torchvision==0.8.2 torchaudio==0.7.2 cudatoolkit=11.0 -c pytorch) because I have CUDA 11.4
I tried both vivos dataset and common voice dataset, store them in .txt with panda seperatated by "|" and 2 column: path (path on server) and transcript (encoded utf-8)
When I tried to print the pred and label, i got these
![image](https://github.com/khanld/ASR-Wav2vec-Finetune/assets/65654745/62335d7c-2d6c-43f8-87ea-5631141229b3)

**Comment by ghosthunterk** [comment]
Audio are already pre-processed to be 16000 sampling rate and .wav format
![image](https://github.com/khanld/ASR-Wav2vec-Finetune/assets/65654745/8630765c-0d42-4798-a7db-c4df80763c84)

**Comment by khanld** [comment]
i can see that your model did not converge yet, train loss is still high. Try increase the lr higher for faster training

**Comment by khanld** [comment]
Ping me at mail khanhld218@gmail.com for better debugging since I rarely check the GitHub notifications

**Comment by ghosthunterk** [comment]
> Ping me at mail [khanhld218@gmail.com](mailto:khanhld218@gmail.com) for better debugging since I rarely check the GitHub notifications

Already, thanks

**Comment by Shaobo-Z** [comment]
Is it possible to get an update on this question? What is the minimum size of the dataset? I want to train the model with a 20mins dataset. Do you think it is possible?

________________________________

**Comment by khanld** [comment]
I will take a look at my codes and run some experiments on english datasets and response to you soon @Shaobo-Z

**Comment by ghosthunterk** [comment]
![image](https://github.com/khanld/ASR-Wav2vec-Finetune/assets/65654745/7df97fdb-8245-4a85-9da8-c541d052f06d)
So after having experimented a while, I found that increasing the learning rate (about >1e-5) and set the scheduler max learning rate to >=1e-4 helped the model to actually learn after a while, just be patient.
